kura/org.eclipse.kura.db.sqlite.provider/OSGI-INF/metatype/org.eclipse.kura.internal.db.sqlite.provider.SqliteDbServiceImpl.xml

```diff
@@ -0,0 +1,91 @@
+    available under the terms of the Eclipse Public License 2.0
+    which is available at https://www.eclipse.org/legal/epl-2.0/
+ 
+	SPDX-License-Identifier: EPL-2.0
+	
+	Contributors:
+	 Eurotech
+
+-->
+<MetaData xmlns="http://www.osgi.org/xmlns/metatype/v1.2.0" localization="en_us">
+    <OCD id="org.eclipse.kura.internal.db.sqlite.provider.SqliteDbServiceImpl" 
+         name="SqliteDbService" 
+         description="Sqlite based database service.">
+
+        <AD id="db.mode"
+            name="Database Mode"
+            type="String"
+            cardinality="0" 
+            required="true"
+            default="IN_MEMORY"
+            description="Defines the database mode.">
+            <Option label="In Memory" value="IN_MEMORY"/>
+            <Option label="Persisted" value="PERSISTED"/>
+         </AD>
+         
+         <AD id="db.path"
+            name="Persisted Database Path"
+            type="String"
+            cardinality="0" 
+            required="true"
+            default="mydb"
+            description="Defines the database path. This parameter is only relevant for persisted databases.">
+         </AD>
+         
+         <AD id="db.journal.mode"
+            name="Journal Mode"
+            type="String"
+            cardinality="0" 
+            required="true"
+            default="WAL"
+            description="The database journal mode (see https://www.sqlite.org/pragma.html#pragma_journal_mode for more details). If set to Rollback Journal the DELETE journal mode will be used. This parameter is only relevant for persisted databases.">
+            <Option label="Rollback Journal" value="ROLLBACK_JOURNAL"/>
+            <Option label="WAL" value="WAL"/>
+         </AD>
+         
+         <AD id="db.defrag.interval.seconds"
+            name="Defrag interval (seconds)"
+            type="Long"
+            cardinality="0" 
+            required="true"
+            default="900"
+            min="60"
+            description="SqliteDbService instances support running periodic defragmentation using the VACUUM command (https://www.sqlite.org/lang_vacuum.html). This parameter specifies the interval in minutes beetween two successive checkpoints, set to zero to disable. This parameter is only relevant for persisted databases."/>
+
+         <AD id="db.wal.checkpoint.interval.seconds"
+            name="WAL Checkpoint Interval (Seconds)"
+            type="Long"
+            cardinality="0" 
+            required="true"
+            default="600"
+            min="0"
+            description="SqliteDbService instances support running periodic periodic WAL checkpoints (https://www.sqlite.org/pragma.html#pragma_wal_checkpoint). Checkpoints will be performed in TRUNCATE mode. This parameter specifies the interval in seconds beetween two consecutive checkpoints, set to zero to disable. This parameter is only relevant for persisted databases in WAL Journal Mode."/>
+
+         <AD id="db.connection.pool.max.size"
```

Comment:
Would it make sense to move this and the following one up in the metatype as it affects both in memory and persistent databases? What is the frequency of usage of such properties for a normal setup?

Reply:
Maybe it is fine as is, the user probably will not change frequently the connection pool size, and the last parameter is just for debug purposes

Reply:
Agreed